### PR TITLE
[Gekidou  MM-45521] Remove show results button

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary

This PR removes the show results button because the filter was converted to a single select option

- remove show results button
- if the user selects the initial filter again, dismiss the modal and don't fetch the data again
- calculate the snap height dynamically and remove magic number
- remove from i18n

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45521

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Screenshots
Before:
image

After: 
image

#### Release Note

```release-note

```

### CODE REVIEW

**Review by matthewbirtch — APPROVED** [review]

**Review by avinashlng1080 — APPROVED** [review]
LGTM - one small correction left

File: app/screens/home/search/results/filter.tsx [hunk 1]
```diff
@@ -98,45 +102,41 @@ const Filter = ({initialFilter, setFilter}: FilterProps) => {
...
+    const handleOnPress = useCallback((fileType: FileFilter) => {
+        if (fileType !== initialFilter) {
+            setFilter(fileType);
+        }
+        dismissBottomSheet();
+    }, [initialFilter]);
+
+    const renderFilterItem = useCallback(({item}) => {
         return (
             <MenuItem
```
Comment (avinashlng1080) [inline app/screens/home/search/results/filter.tsx:125]: Replace MenuItem with OptionItem

File: app/screens/home/search/results/header.tsx [hunk 2]
```diff
@@ -78,6 +82,15 @@ const Header = ({
         onTabSelect('files');
     }, [onTabSelect]);
 
+    const snapPoints = useMemo(() => {
+        return [
+            bottomSheetSnapPoint(
+                NUMBER_FILTER_ITEMS + 1,
+                FILTER_ITEM_HEIGHT,
+                bottom) + TITLE_HEIGHT,
```
Comment (larkox) [inline app/screens/home/search/results/header.tsx:90]: I am not a big fan of the indentation here, maybe like this?
```suggestion
            bottomSheetSnapPoint(
                NUMBER_FILTER_ITEMS + 1,
                FILTER_ITEM_HEIGHT,
                bottom,
            ) + TITLE_HEIGHT,
```

**Review by larkox — APPROVED** [review]
LGTM, just one small nit.

**Review by avinashlng1080 — APPROVED** [review]

**Pull request merged**